pos_order_mgmt/static/src/js/widgets.js

```diff
@@ -0,0 +1,450 @@
+
+            var search_timeout = null;
+            this.$('.searchbox input').on('keyup', function (event) {
+                self.search_query = this.value;
+                clearTimeout(search_timeout);
+                search_timeout = setTimeout(function () {
+                    self.perform_search();
+                }, 70);
+            });
+
+            this.$('.searchbox .search-clear').click(function () {
+                self.clear_search();
+            });
+
+            this.perform_search();
+        },
+
+        render_list: function () {
+            var self = this;
+            var orders = this.orders;
+            var contents = this.$el[0].querySelector('.order-list-contents');
+            contents.innerHTML = "";
+            for (var i = 0, len = Math.min(orders.length, 1000); i < len; i++) {
+                var order = orders[i];
+                var orderline = this.order_cache.get_node(order.id || order.uid);
+                if (!orderline) {
+                    var orderline_html = QWeb.render('OrderLine', {
+                        widget: this,
+                        order: order,
+                    });
+                    orderline = document.createElement('tbody');
+                    orderline.innerHTML = orderline_html;
+                    orderline = orderline.childNodes[1];
+                    this.order_cache.cache_node(order.id || order.uid, orderline);
+                }
+                if (order === this.old_order) {
+                    orderline.classList.add('highlight');
+                } else {
+                    orderline.classList.remove('highlight');
+                }
+                contents.appendChild(orderline);
+            }
+            // FIXME: Everytime the list is rendered we need to reassing the
+            // button events.
+            this.$('.order-list-return').off('click');
+            this.$('.order-list-reprint').off('click');
+            this.$('.order-list-copy').off('click');
+            this.$('.order-list-return').click(function (event) {
+                self.order_list_actions(event, 'return');
+            });
+            this.$('.order-list-reprint').click(function (event) {
+                self.order_list_actions(event, 'print');
+            });
+            this.$('.order-list-copy').click(function(event) {
+                self.order_list_actions(event, 'copy');
+            });
+        },
+
+        order_list_actions: function (event, action) {
+            var self = this;
+            var order_data;
+            var dataset = event.target.parentNode.dataset;
+            if (dataset.orderId) {
+                self.load_order_data(parseInt(dataset.orderId, 10))
+                .then(function(order_data) {
+                    self.order_action(order_data, action);
+                })
+            } else {
+                _.each(this.orders, function (order_data) {
+                    if (order.uid === dataset.uid) {
```

Comment:
I don't understand in which case dataset.orderId is not defined, and order is not defined anywhere.